packages/gatsby-remark-katex/package.json

```diff
@@ -8,7 +8,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "katex": "^0.9.0",
+    "katex": "^0.10.0",
```

Comment:
Let's make this a peerDependency like we do in other plugins, e.g. `gatsby-remark-prismjs`, `gatsby-plugin-emotion`, etc.

That way users can install whatever version they'd like, rather than mandating changes like these. I'd recommend something like

```json
"peerDependencies": {
  "katex": ">=0.10.0 <1.0.0"
}
```

(<1.0.0 versions are weird with semver, e.g. `^0.10.0` does not satisfy `0.11.0`)

Reply:
@DSchau Thanks for the tips! Should I also move `unist-util-visit` and `remark-math` to `peerDependencies`?

Reply:
Just Katex for now, please!

Reply:
@DSchau  Thanks for the clarification! I am not very familiar with dealing with peerDependencies, I see if I `yarn` it when peer depended, the reference to `katex` inside `yarn.lock` is removed. Is this expected?

Also should I rename this PR? I see many PRs here have names like `feat(Gatsby): `, `chores(Gatsby): ` etc., but couldn't find a full description of the naming convention used. Should this PR be called `update(gatsby-remark-katex)` or similar? :)

Reply:
@hexrcs thanks for noticing! We usually do that come merge time, but I went ahead and did it for you. The general idea is that the general _spirit_ of the PR (e.g. new feature, bug fix, chore, etc.) is to be used, so we can tie into conventional changelog stuff!

> I am not very familiar with dealing with peerDependencies, I see if I yarn it when peer depended, the reference to katex inside yarn.lock is removed. Is this expected?

Yup! So you'll want to do something like `yarn add katex --peer` which appears to be what you did. That should work just fine!

Reply:
@DSchau Awesome, thanks! Will keep that in mind next time I contribute 😛

Updated this PR 😃

Reply:
We should actually _document_ it :) Because we'd love it if contributors would do that themselves, so seriously thanks so much for noticing and being aware of it 💜 

If you're interested in taking a stab at that document that would be 💯 too! It should probably be documented [here](https://www.gatsbyjs.org/docs/how-to-contribute/), and then maybe in the pull request template, as well.

No pressure at all though, we're already so glad you contributed!

Reply:
> If you're interested in taking a stab at that document that would be 💯 too! It should probably be documented [here](https://www.gatsbyjs.org/docs/how-to-contribute/), and then maybe in the pull request template, as well.

Sure I'll look in to that! Glad I could help, I love Gatsby! 😄